### PR TITLE
Fixes a bug with question select widget in report builder

### PR DESCRIPTION
**Pull request by NoahCarnahan**
@nickpell cc @kaapstorm 
related to https://github.com/dimagi/commcare-hq/pull/8397

### CODE REVIEW

**Comment by NoahCarnahan** [comment]
Doesn't actually fix the issue

**Comment by NoahCarnahan** [comment]
ok fixed

**Comment by nickpell** [comment]
nice

File: corehq/apps/userreports/reports/builder/forms.py [hunk 1]
```diff
@@ -102,19 +102,25 @@ def render(self, name, value, attrs=None, choices=()):
...
+            """,
             flatatt(final_attrs),
             mark_safe(self.render_options(choices)),
             value
         )
 
     def render_options(self, choices):
+
+        def escape(literal):
+            return literal.replace('&', '&amp;').replace("'", "&#39;")
```
Comment (czue) [inline corehq/apps/userreports/reports/builder/forms.py]: is there a reason not to use something like `cgi.escape(literal, quote=True)` here?

Comment (NoahCarnahan) [inline corehq/apps/userreports/reports/builder/forms.py]: I definitely would prefer a library function, but I couldn't find one that does quite what I want (although I suspect one exists). This string is going to be used in the above single-quoted data-bind attribute. Based on the [HTML spec](http://www.w3.org/TR/html5/syntax.html#attribute-value-%28single-quoted%29-state), it looks like ampersand and single quote are the only characters that need escaping in the "attribute value (single-quoted) state".
(`cgi.escape` escapes `&`, `<`, `>`, and optionally `"`, but not `'`)

Comment (kaapstorm) [inline corehq/apps/userreports/reports/builder/forms.py]: Yeah, I'm guessing if you need to escape "&", then you probably need to escape other characters too. I think `cgi.escape` will escape double quotes but not single quotes. `django.utils.html.escape` will escape both kinds of quotes. That's probably what you want, or you could use `cgi.escape` and switch your outer quotes to double quotes, i.e.

    """
    <input{0} data-bind="
        questionsSelect: {1},
        value: '{2}',
        optionsCaption: ' '
    "/>
    """,

**Pull request merged**